frontend/src/hooks/__tests__/jest/useDeAuthenticate.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { useDeAuthenticate } from "@/hooks/useDeAuthenticate";
+import { useContext as useContextMock } from "react";
+import { useRouter as useRouterMock } from "next/router";
+import { authenticationContextDefaultValue } from "@/contexts/AuthenticationContext";
+
+jest.mock("react", () => ({
+  ...jest.requireActual("react"),
+  useContext: jest.fn(),
+  useCallback: <T>(fn: T): T => fn,
+}));
+
+jest.mock("next/router", () => ({
+  useRouter: jest.fn(),
+}));
+
+const useContext = useContextMock as jest.Mock;
+const useRouter = useRouterMock as jest.Mock;
+
+const updateAuthenticationContext = jest.fn();
+const routerPush = jest.fn();
+
+beforeEach(() => {
+  jest.resetAllMocks();
+
+  useContext.mockReturnValue(updateAuthenticationContext);
+  useRouter.mockReturnValue({ push: routerPush });
+});
+
+describe("useDeAuthenticate", () => {
+  it("should reset the authenticationstate", () => {
+    const deAuthenticate = useDeAuthenticate();
```

Comment:
In general, `deauthenticate` is one word, so `Deauthenticate` would be the proper capitalization.

On the other hand, `logout` (respectively `useLogout`) might be easier to read and shorter, although admittedly less accurate if we wanted to be pedantic. I would still suggest this change based on the number of times I tripped over deauthenticated when reviewing, and mixing it with authenticate (similar length, similar ending) 😅